### ISSUE TITLE
Early error with an invalid regular expression flag

### ISSUE THREAD

**Issue by igor-simoes** [facebook/hermes#268]
Hi everyone,
hermes should throw an early SyntaxError instead of a ReferenceError with an invalid regular expression flag. The same occurs using eval.

version: 0.5.0
OS: Linux Mint 19.2 x64

Steps to reproduce:
```
b+/v/a
```

Actual results:
```
ReferenceError: Property 'b' doesn't exist
```

Expected results:
```
SyntaxError: Invalid regular expression flags
```

Removing the `b` variable, the SyntaxError was thrown as expected
```
+/v/a
SyntaxError: Invalid flags passed to RegExp
```

V8, SpiderMonkey, Chakra and JavaScriptCore works as expected.

cinfuzz

**Comment by tmikov** [comment]
Thank you for reporting this. It appears we are not doing early validation of the regexp flags.

**Comment by neildhar** [comment]
Hi @igor-simoes, thank you for reporting this bug! It is now fixed.

**Comment by avp** [comment]
Fixed in db57e8c9033abed1ccb9de4df135d9a2d67a38e2
